drivers/pure/flasharray/pods.go

```diff
@@ -0,0 +1,57 @@
+package flasharray
+
+import "github.com/portworx/torpedo/pkg/log"
+
+type PodServices struct {
+	client *Client
+}
+
+func (p *PodServices) CreatePod(params map[string]string, data interface{}) (*[]PodResponse, error) {
+	req, _ := p.client.NewRequest("POST", "pods", params, data)
+	m := &[]PodResponse{}
+	_, err := p.client.Do(req, m)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+func (p *PodServices) ListAllAvailablePods(params map[string]string, data interface{}) ([]PodResponse, error) {
+	req, err := p.client.NewRequest("GET", "pods", params, data)
+	if err != nil {
+		return nil, err
+	}
+	m := []PodResponse{}
+	_, err = p.client.Do(req, &m)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+func (p *PodServices) PatchPod(Patchparams map[string]string, data interface{}) ([]PodResponse, error) {
```

Comment:
[nit] parameter names should be camelCase
```suggestion
func (p *PodServices) PatchPod(patchParams map[string]string, data interface{}) ([]PodResponse, error) {
```